### PR TITLE
Optimise packaging

### PR DESCRIPTION
**Pull request by viktorsaws**
## Problem

1. We use deprecated unsecure `vercel/pkg` package and mention it as main way of consuming bundled runtime servers, which also takes a lot of time (3-4 minutes per run)
2. Our github workflows take ~7 minutes in Linux and ~13 minutes on Windows runners to complete. Npm commands called in workflows are wasteful: npm commands in workflow call `npm install` and `npm run compile` under the hood 2 or 3 times, which adds up to workflow run times.

## Solution

To reduce workflow run times:
1. Removed using pkg for producing binaries
2. Removed redundant npm script calls during workflow runs

## Result
Run time for all workflows is 3x times faster now, taking ~4 minutes.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ege0zcan** [comment]
Nice change, should we also consider renaming the folder names from `-binary` to something like `-standalone` or something else

File: app/aws-lsp-json-binary/package.json [hunk 1]
```diff
@@ -8,9 +8,9 @@
     },
     "scripts": {
         "compile": "tsc --build",
-        "package-x64": "pkg --targets node18-linux-x64,node18-win-x64,node18-macos-x64 --out-path bin --compress GZip .",
+        "pkg": "pkg --targets node18-linux-x64,node18-win-x64,node18-macos-x64 --out-path bin --compress GZip .",
```
Comment (ege0zcan) [inline app/aws-lsp-json-binary/package.json:11]: so we still want to keep this usage of pkg right?

File: app/aws-lsp-yaml-json-webworker/package.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
         "compile": "tsc --build",
         "clean": "rm -rf dist && rm -rf node_modules && rm -rf build",
         "build": "tsc",
-        "prepublishOnly": "npm run build && npm run compile && npm run bundle",
-        "bundle": "webpack",
+        "prepublishOnly": "npm run build && npm run compile && npm run package",
+        "package": "webpack",
```
Comment (ege0zcan) [inline app/aws-lsp-yaml-json-webworker/package.json:10]: do we want to keep the `package` name in the scripts, why not rename all usages of it with something like `bundle` instead?

Comment (ege0zcan) [reply app/aws-lsp-yaml-json-webworker/package.json:10]: is it because we still have a few subpackages that use `pkg` and do actual packaging?

**Comment by ege0zcan** [comment]
We also need to update integration tests that make use of executables since they will no longer work: 

https://github.com/aws/language-servers/blob/main/app/hello-world-lsp-runtimes/src/tests/helloWorldServerInteg.test.ts#L20-L40

we should instead spawn a child process like `node /build/server.js`

File: app/aws-lsp-json-binary/package.json (hunk 1 above)
Comment (ege0zcan) [reply app/aws-lsp-json-binary/package.json:11]: I think if we are only keeping it in for testing, we can remove these and use the bundled js files for testing

File: app/hello-world-lsp-runtimes/README.md [hunk 3]
```diff
@@ -24,26 +24,17 @@ npm run bundle
...
-    ```
-
-* To create only javascript bundled files with webpack, run:
+* To create javascript bundled files with webpack, run:
     ```bash
     npm run webpack
     ```
 
-Result artifacts can be found in `/out` and `/bin` directories.
+Result artifacts can be found in `/out` and `/build` directories.
```
Comment (ege0zcan) [inline app/hello-world-lsp-runtimes/README.md:32]: nit: the following file tree doesn't show the build folder

File: app/aws-lsp-yaml-json-webworker/package.json (hunk 2 above)
Comment (viktorsaws) [reply app/aws-lsp-yaml-json-webworker/package.json:10]: I wanted to keep naming consistent with older variants, I don't think it matters much and we used `package` for most of such commands

**Review by ege0zcan — APPROVED** [review]

File: app/aws-lsp-json-runtimes/src/tests/jsonServerCFInteg.test.ts [hunk 4]
```diff
@@ -20,25 +19,12 @@ describe('Test JsonServer with CloudFormation schema', () => {
     let process: ChildProcessWithoutNullStreams
     let endpoint: JSONRPCEndpoint
     let client: LspClient
-    let binaryFile = 'lsp-json-binary'
+    // let runtimeFile = 'lsp-json-runtimes'
```
Comment (ege0zcan) [inline app/aws-lsp-json-runtimes/src/tests/jsonServerCFInteg.test.ts:22]: nit: leftover comment

**Review by ege0zcan — APPROVED** [review]

**Pull request merged**